backend/director/agents/web_search_agent.py

```
@@ -0,0 +1,172 @@
+
+    def _handle_video_search(self, query: str, count: int, duration: str, collection_id: str) -> AgentResponse:
+        """Handles video searches."""
+        videos_content = VideosContent(
+            agent_name=self.agent_name,
+            status=MsgStatus.progress,
+            status_message="Searching for videos...",
+            videos=[],
+        )
+
+        for _ in range(count):
+            video_data = VideoData(
```

Ruff (0.8.2)

Undefined name `VideoData`

(F821)